eppo_client/read_write_lock.py

```diff
@@ -1,6 +1,7 @@
 import threading
+from contextlib import contextmanager
 
-# Copied from: https://www.oreilly.com/library/view/python-cookbook/0596001673/ch06s04.html
+# Adapted from: https://www.oreilly.com/library/view/python-cookbook/0596001673/ch06s04.html
```

Comment:
Show me the LLM vectors

Reply:
I tried that first, it went down a much more complicated path so I just added the contextmanager myself :)